### PR TITLE
Event Feed project filtering for the GetFeed request

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Adding project filtering for the GetFeed endpoint event-feed-service. This is one of the three endpoints that need project filtering. Project filtering is only used for chef-server events which are not added yet. 

### :chains: Related Resources

https://github.com/chef/automate/issues/444

### :athletic_shoe: How to Build and Test the Change
This might be hard to test. 
1. `build event-feed-service && start_all_services`
1. Add a project named "project9"
1. Add a project named "project3"
1. Add a chef-server event tagged with "project9" run `event_feed_add_event` 
1. Create a profile event by adding a profile from the "Available" section here https://a2-dev.test/compliance/compliance-profiles
1. Request the feed for project3 only. Ensure that the chef-server event is not returned, but the profile event is returned
1. Request the feed for projects9 only. Ensure both events are returned. 

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/config-mgmt-service/backend/elastic/elastic.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func newBoolQueryFromFilters(filters map[string][]string) *elastic.BoolQuery {
 			refinedValues = stringutils.SliceFilter(values, func(projectId string) bool {
 				return projectId != authzConstants.UnassignedProjectID
 			})
-		} else { // Do not want projects to use wildcards
```
Comment (lancewf) [inline components/config-mgmt-service/backend/elastic/elastic.go:115]: We do not want wildcards for project filtering

File: components/event-feed-service/pkg/integration_test/suite_test.go [hunk 2]
```diff
@@ -69,13 +70,20 @@ func NewSuite(url string) (*Suite, error) {
...
+
 	if err != nil {
-		return nil, errors.Wrap(err, "initializing gRPC clients")
+		return nil, errors.Wrap(err, "loading hab grpc conn factory")
 	}
-	s.feedClient = feedClient
-	s.purgeClient = purgeClient
-	s.cleanup = cleanup
+
+	s.feedClient = server.New(s.feedBackend)
```
Comment (lancewf) [inline components/event-feed-service/pkg/integration_test/suite_test.go:84]: These changes allow the local changes to be tested and not the running service.

File: components/event-feed-service/pkg/persistence/feed-elastic.go [hunk 3]
```diff
@@ -150,13 +153,23 @@ func (efs ElasticFeedStore) GetFeed(query *feed.FeedQuery) ([]*feed.FeedEntry, i
 		mainQuery = mainQuery.Must(rangeQuery)
 	}
 
-	searchService := efs.client.Search().
+	searchSource := olivere.NewSearchSource().
```
Comment (lancewf) [inline components/event-feed-service/pkg/persistence/feed-elastic.go:156]: These changes allow the elasticsearch query to be logged for testing.

File: components/event-feed-service/pkg/persistence/feed-elastic.go [hunk 4]
```diff
@@ -228,15 +241,50 @@ func newBoolQueryFromFilters(filters map[string][]string) *olivere.BoolQuery {
...
+				}
+			}
+		}
 
+		// Even if there is a wildcard value found, we still want to narrow down by any other values.
+		// This would probably negate anything found with wildcards but using should brings back extra results
+		if len(refinedValues) > 0 {
+			termQuery := olivere.NewTermsQuery(field, stringArrayToInterfaceArray(refinedValues)...)
+			filterQuery = filterQuery.Should(termQuery)
+		}
```
Comment (lancewf) [inline components/event-feed-service/pkg/persistence/feed-elastic.go:277]: This adds project filters and wild card filtering.

File: components/event-feed-service/pkg/persistence/mappings.go [hunk 5]
```diff
@@ -133,6 +133,17 @@ var feedProps = `
...
 			"format":"strict_date_optional_time||epoch_millis"
+		},
+		"chef_organization": {
+			"type": "keyword",
+			"ignore_above": 256
+		},
+		"chef_infra_server": {
+			"type": "keyword",
+			"ignore_above": 256
+		},
```
Comment (lancewf) [inline components/event-feed-service/pkg/persistence/mappings.go:144]: We need the "chef_organization" and "chef_infra_server" to perform project updates.

File: .studio/event-feed-service [hunk 6]
```diff
@@ -21,8 +21,56 @@ document "event_feed_add_event" <<DOC
...
+document "generate_random_event_example" <<DOC
+  Add an example event with random ID and time from the last 144 hours.
+DOC
+function generate_random_event_example() {
+  local uuid
+  uuid=$(uuidgen)
+  local rfc_time
+  rfc_time=$(date +%FT%TZ -d "$((RANDOM % 144)) hour ago")
+
+  generate_event_example | jq --arg uuid "$uuid" --arg rfc_time "$rfc_time" '.EventID = $uuid | .Published = $rfc_time'
```
Comment (danielsdeleo) [inline .studio/event-feed-service:73]: Thanks for adding the composable style sample data generators :)

File: components/event-feed-service/pkg/persistence/feed-elastic.go (hunk 3 above)
Comment (danielsdeleo) [reply components/event-feed-service/pkg/persistence/feed-elastic.go:156]: ooh, this is way nicer than a 100 line comment with an example

File: .studio/event-feed-service (hunk 6 above)
Comment (lancewf) [reply .studio/event-feed-service:73]: I copied off you. :)

**Pull request merged**